homeassistant/components/firmata/config_flow.py

```diff
@@ -0,0 +1,99 @@
+"""Config flow to configure firmata component."""
+
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_HOST, CONF_NAME
+from homeassistant.core import callback
+
+from .const import DOMAIN, CONF_SERIAL_PORT, CONF_PORT
+
+
+@callback
+def configured_boards(hass):
+    """Return a set of all configured boards."""
+    return {entry.data[CONF_NAME]: entry for entry
+            in hass.config_entries.async_entries(DOMAIN)}
+
+
+@config_entries.HANDLERS.register(DOMAIN)
+class FirmataFlowHandler(config_entries.ConfigFlow):
+    """Handle a firmata config flow."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_LOCAL_PUSH
+
+    _hassio_discovery = None
+
+    def __init__(self):
+        """Initialize the firmata config flow."""
+        self.firmata_config = {}
+
+    async def async_step_init(self, user_input=None):
+        """Needed in order to not require re-translation of strings."""
+        return await self.async_step_user(user_input)
+
+    async def async_step_user(self, user_input=None):
+        """Handle a firmata config flow start.
+
+        If only one board is found go to link step.
+        If more than one board is found let user choose board to link.
+        If no board is found allow user to manually input configuration.
+        """
+        if user_input is not None:
+            self.firmata_config = user_input
+            return await self._create_entry()
```

Comment:
I never got this form to work, but I also did not implement the entire config feature set. Is there a way to implement config entries without the forms at all?

Reply:
What do you want to do?

Reply:
Right now the config entries work but not the config flow. The component can be configured from `configuration.yaml` but not from the UI. While I'm sure it would be possible to implement, it's a lot of config options.

Reply:
We only want the critical options in the config flow. Extra options should stay in configuration.yaml.